### ISSUE TITLE
Be able to send a file in an HttpResponse

### ISSUE THREAD

**Issue by LoicPoullain** [FoalTS/foal#374]
# Issue

Currently there is no easy way to download a file from a Foal application. A workaround is to completely load the file in the process and then send it to the client.

```typescript
import { promisify } from 'util';
import { readFile } from 'fs';

export class AppController {

  @Get('*')
  async notFound() {
    const index = await promisify(readFile)('./public/index.html', 'utf8');
    return new HttpResponseOK(index);
  }

}
```

It is not ideal because:
- it slows the response time,
- it unnecessarily increases the memory used by the node process,
- and it apparently only works with text files. The `Content-Type` of the response currently cannot be set.

# Solution

One solution could be to allow to send a file in streaming using a `stream: true` option:
```typescript
import { createReadStream } from 'fs';

// FoalTS
const rstream = createReadStream('./public/index.html');
return new HttpResponseOK(rstream, { stream: true });

// Implementation under the hood in @foal/core
const rstream = createReadStream('./public/index.html');
rstream.pipe(res); // `res` is the express response.
```

This would be only a part of the solution though since it does not solve the `Content-Type` problem.

**Comment by LoicPoullain** [comment]
Resolved in v0.8.7

**Comment by Paul75** [comment]
Hello, I want to send PDF to Angular. I use pdf-lib butr not seams to works ...

My function : 

```
const pdfBytes = await pdfDoc.save();

return new HttpResponseOK(pdfBytes)
      .setHeader('Content-Type', 'application/pdf');
```

But the file get : 

`{"0":37,"1":80,"2":68,"3":70,"4":45,"5":49,"6":46,"7":55,"8":10,"9":37,
`
Thanks
